### DISCUSSION TITLE
Understanding the Optimization Process of next/image in Next.js

### DISCUSSION THREAD

**Discussion by samiur-r** [vercel/next.js#55747] [Help] [answered]
### Summary

I'm currently working with Next.js and the next/image component for image rendering. I've been using it to optimize image loading in my application, but I'm curious about the technical details of how it works under the hood. Specifically, I'd like to know if it makes any API calls during the optimization process.

Here's what I understand so far:

- next/image provides features like lazy loading, responsive images, and format selection to optimize image loading.
- It generates multiple versions of images in different sizes and formats during the build process.
- It selects the most suitable image format based on browser support.
- It supports lazy loading by default, and you can specify responsive image dimensions.

However, I couldn't find information on whether next/image makes any API calls during this optimization process. My goal is to gain a deeper understanding of its internal workings. Any insights or resources on this topic would be greatly appreciated.

**Comment by icyJoseph** [comment, marked as answer]
You can see in the HTML that a `next/image` url points to `_next/image`.

In the server it uses something called `sharp`, if available, otherwise, it uses something called `squoosh`:

- https://www.npmjs.com/package/sharp
- https://github.com/vercel/next.js/blob/fabab8cf05da106410d97d818adf2f6b4742d614/packages/next/src/server/image-optimizer.ts

In the server, the images are optimised by sharp.

In the client the next/image components takes care of sizes and srcset, so that the browser loads an image size that fits media.

**Reply by samiur-r** [reply]
Thank you for the insights

**Comment by samiur-r** [comment]
I'm now curious about the image optimization service provided by Vercel (https://vercel.com/docs/image-optimization/managing-image-optimization-costs). If I host my Next.js application on Vercel, does it automatically utilize this image optimization service for the images generated by the next/image component?

I ask because I'm planning to launch a web app with a substantial number of images, possibly in the thousands. Understanding whether Vercel's image optimization service is being used will help me calculate potential costs and optimize my budget accordingly.

Any insights or information on how Vercel's image optimization service integrates with Next.js would be greatly appreciated. Thank you!

**Reply by icyJoseph** [reply]
Ah that's a good question. 

I see on my dashboard/usage page that I do have 4% used up image optimisation quota.

I guess you are bound by these:

- https://vercel.com/docs/limits/fair-use-policy#commercial-usage
- https://vercel.com/docs/limits/fair-use-policy#typical-monthly-usage-guidelines

According to the second link, you can do up to 1000 image optimisations per month, within the Hobby plan, and 5000 as pro.

If your app will have thousands of images, then it looks like you'd have little room to work. Maybe consider another service? The Next.js image API allows for other loaders: 

- https://nextjs.org/docs/app/api-reference/components/image#loader
- Examples: https://nextjs.org/docs/app/api-reference/next-config-js/images

Also, I wonder, because there's caching involved, you maybe get +1 per image URL, rather than per access.

**Reply by samiur-r** [reply]
Once more, I appreciate your valuable insights.
